### PR TITLE
542 xarray

### PR DESCRIPTION
**Pull request by Lukas113**
See issue #542 

Additional bug fixing and minor improvements

Unfortunately, I created this branch from 512_sarus and not from main, if you wonder why there are so many commits.

### CODE REVIEW

**Review by lmachadopolettivalle — APPROVED** [review]
LGTM 💯

**Comment by Lukas113** [comment]
@kenfus is it okay to remove `guess_beam_parameters` in this PR? I removed it because it never worked. If you have a good reason to keep it and ensure that it works, I'll happily add the function to the module again.

File: conda/meta.yaml [hunk 1]
```diff
@@ -55,7 +55,7 @@ requirements:
     - ska-sdp-datamodels      =0.1.3=*_0
     - ska-sdp-func-python     =0.1.4=*_0
     - tools21cm               =2.0.2=*_0
-    - xarray                  >=2022.10.0
+    - xarray                  <=2023.2  # see issue #542
```
Comment (sfiruch) [inline conda/meta.yaml:58]: Please fix version instead.

File: environment.yaml [hunk 2]
```diff
@@ -36,7 +36,7 @@ dependencies:  # package-version & build-number of Karabo-Feedstock deps should
   - ska-sdp-datamodels      =0.1.3=*_0
   - ska-sdp-func-python     =0.1.4=*_0
   - tools21cm               =2.0.2=*_0
-  - xarray                  >=2022.10.0
+  - xarray                  <=2023.2  # see issue #542
```
Comment (sfiruch) [inline environment.yaml:39]: Please fix version instead.

**Comment by Lukas113** [comment]
An update, with pandas >=2, the following FutureWarning occured:

```
/home/lukas/miniconda3/envs/karabo_dev_env/lib/python3.9/site-packages/ska_sdp_datamodels/visibility/vis_model.py:74: FutureWarning: the `pandas.MultiIndex` object(s) passed as 'baselines' coordinate(s) or data variable(s) will no longer be implicitly promoted and wrapped into multiple indexed coordinates in the future (i.e., one coordinate for each multi-index level + one dimension coordinate). If you want to keep this behavior, you need to first wrap it explicitly using `mindex_coords = xarray.Coordinates.from_pandas_multiindex(mindex_obj, 'dim')` and pass it as coordinates, e.g., `xarray.Dataset(coords=mindex_coords)`, `dataset.assign_coords(mindex_coords)` or `dataarray.assign_coords(mindex_coords)`.
  super().__init__(data_vars, coords=coords, attrs=attrs)
  ```

File: karabo/sourcedetection/result.py [hunk 3]
```diff
@@ -110,7 +110,7 @@ def detect_sources_in_image(
         cls: Type[_SourceDetectionResultType],
         image: Image,
         beam: Optional[BeamType] = None,
-        quiet: bool = False,
+        verbose: bool = False,
```
Comment (sfiruch) [inline karabo/sourcedetection/result.py:113]: Did you intend to change the default value?

(Asking because `verbose` is the opposite of `quiet`)

**Review by sfiruch — CHANGES_REQUESTED** [review]

File: karabo/sourcedetection/result.py (hunk 3 above)
Comment (Lukas113) [reply karabo/sourcedetection/result.py:113]: I changed from quiet to verbose to not have different keywords for the same thing, even they're inverted.

The change to be not verbose is intentional, but based on personal preference. I just don't like default verbose output unless the process always takes a large amount of time (there, a `tqdm` bar would be nice).

**Comment by Lukas113** [comment]
I also removed the build-string fixations, because they prevent updates on dependency-constraints. Therefore, I introduces the following instructions to the README of the Feedstock & triggered the according builds there.

**Updating an already existing Wheel**: It may happen that an update of a dependency (direct or transversal) introduces a breaking change in a future release. Then, the current constraints of an existing conda-wheel would be outdated an will most likely result in a broken environment. To prevent this, updating the according meta.yaml and increase the build-nr is not enough, because the solver doesn't just consider the largest build-numbers of conda-wheels. So, to fix this for current and also older releases, you have to relabel oudated conda-wheels to `legacy`. Also do a comment in the meta.yaml, why you did constrain the dependency. This makes future builds where the constraints may have to be reconsidered way easier. It was already mentioned in this paragraph, but don't forget to increase the build-number.

**Review by sfiruch — DISMISSED** [review]

**Comment by kenfus** [comment]
> @kenfus is it okay to remove `guess_beam_parameters` in this PR? I removed it because it never worked. If you have a good reason to keep it and ensure that it works, I'll happily add the function to the module again.

I rather not. This was requested by @rohitcbscient and it also works in the current example of the `source_detection.ipynb`.

**Review by sfiruch — APPROVED** [review]

**Pull request merged**